### PR TITLE
#954 Add unit tests to test scenarios for requests with/without a token

### PR DESCRIPTION
**Pull request by kitsook**
For #954 
- added test case for token handling
- since #750 is not fully resolved, the test case is set as ignored and puzzle added

### CODE REVIEW

**Comment by karato** [comment]
@kitsook Thanks, someone will review your pull request soon

**Comment by karato** [comment]
@pinaf do this review

**Comment by pinaf** [comment]
@karato please assign someone else

**Comment by karato** [comment]
> @karato please assign someone else

@pinaf -15 points to your rating :(

**Comment by karato** [comment]
> @karato please assign someone else

@pinaf right, I'll try to find someone else for this task

**Comment by karato** [comment]
@krzyk review it

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 1]
```diff
@@ -58,6 +60,9 @@
  *  should be added, e.g. when a friend is not found, the error
  *  should be reported correctly and bout creation should be
  *  discarded.
+ * @todo #750:30min TkStart should not create new bout
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook you should set correct issue, here it should be #960 (the one you are solving)

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 2]
```diff
@@ -150,7 +155,7 @@ public void renamesBout() throws Exception {
                         String.format(
                             "/start?post=message&invite=%s&rename=%s",
                             base.randomAlias().name(),
-                            URLEncoder.encode(name, "UTF-8")
+                            URLEncoder.encode(name, CharEncoding.UTF_8)
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook `StandardCharsets.UTF_8.name()`

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 3]
```diff
@@ -238,4 +243,72 @@ public void handlesMultiplePosts() throws Exception {
...
+                new TkForward(new TkStart(base)).act(
+                    new RqWithAuth(
+                        urn,
+                        new RqFake(
+                            RqMethod.GET,
+                            String.format(
+                                "/start?post=%s&token=%s",
+                                URLEncoder.encode(
+                                    msgs[count],
+                                    CharEncoding.UTF_8
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook StandardCharsets.UTF_8.name()

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 4]
```diff
@@ -238,4 +243,72 @@ public void handlesMultiplePosts() throws Exception {
...
+                                ),
+                                URLEncoder.encode(
+                                    tokens[count],
+                                    CharEncoding.UTF_8
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook StandardCharsets.UTF_8.name()

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 5]
```diff
@@ -238,4 +243,72 @@ public void handlesMultiplePosts() throws Exception {
...
+            "Hello World",
+            "I shouldn't be created as new",
+            "I should be created as new",
+            "I am new too",
+            "Another new",
+        };
+        final String[] tokens = {
+            // @checkstyle MultipleStringLiteralsCheck (2 lines)
+            "13579",
+            "13579",
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook make it a variable and reuse it two times

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 6]
```diff
@@ -58,6 +60,9 @@
  *  should be added, e.g. when a friend is not found, the error
  *  should be reported correctly and bout creation should be
  *  discarded.
+ * @todo #750:30min TkStart should not create new bout
+ *  when same token is specified. Currently set the test case
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook you should mention which method should be unignored when this todo is done

**Comment by krzyk** [comment]
@kitsook please see my comments

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java (hunk 1 above)
Comment (kitsook) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @krzyk #960 is a PR. ok to reference a PR instead the issue?

**Comment by kitsook** [comment]
@krzyk revised. please check

File: netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java [hunk 7]
```diff
@@ -58,6 +60,9 @@
  *  should be added, e.g. when a friend is not found, the error
  *  should be reported correctly and bout creation should be
  *  discarded.
+ * @todo #954:30min TkStart should not create new bout
+ *  when same token is specified. Currently set the test case
```
Comment (krzyk) [inline netbout-web/src/test/java/com/netbout/rest/bout/TkStartTest.java]: @kitsook maybe rather: `When this is fixed, remove @Ignore from handlesTokens test`.

**Comment by krzyk** [comment]
@kitsook one comment regarding the todo

**Comment by kitsook** [comment]
@krzyk revised. please check

**Comment by krzyk** [comment]
@kitsook looks good

**Comment by krzyk** [comment]
@rultor merge pls

**Comment by dmzaytsev** [comment]
@rultor merge

**Comment by karato** [comment]
@elenavolokhova please, check this issue for QA compliance, as per [par.24](http://at.teamed.io/policy.html#24)

**Comment by karato** [comment]
@rultor please deploy

**Comment by elenavolokhova** [comment]
@karato Looks good!

**Comment by karato** [comment]
> @karato Looks good!

@elenavolokhova thanks a lot

**Comment by karato** [comment]
@krzyk I added 10 mins to @elenavolokhova (for QA review) in transaction 75351337

10 mins added to @dmzaytsev account (our architect), payment ID is `75351355`, since quality is good here

Much appreciated! **26 mins** was added to your account, payment ID `75351375`, time spent is 22 hours and 47 mins

review comments (c=11) added as a bonus

+26 added to your rating, at the moment it is: [+4213](http://www.netbout.com/b/35035?open=rating)

**Pull request merged**